htdocs/admin/content/alliance/edit.inc.php

```diff
@@ -1,638 +1,534 @@
 <?PHP
 
-		if (isset($_GET['alliance_id']))
-			$id = $_GET['alliance_id'];
-		if (isset($_GET['id']))
-			$id = $_GET['id'];
-
-
-		if (isset($_POST['info_save']) && $_POST['info_save']!="")
-		{
-			//  Bild löschen wenn nötig
-			$img_sql="";
-			if (isset($_POST['alliance_img_del']))
-			{
-				$res = dbquery("SELECT alliance_img FROM alliances WHERE alliance_id=".$id.";");
-				if (mysql_num_rows($res)>0)
-				{
-					$arr=mysql_fetch_array($res);
-					if (file_exists('../'.ALLIANCE_IMG_DIR."/".$arr['alliance_img']))
-					{
-						unlink('../'.ALLIANCE_IMG_DIR."/".$arr['alliance_img']);
-					}
-					$img_sql=",alliance_img=''";
-				}
+use EtoA\Alliance\AllianceBuildingRepository;
+use EtoA\Alliance\AllianceRepository;
+use EtoA\Alliance\AllianceTechnologyRepository;
+
+/** @var AllianceRepository */
+$repository = $app['etoa.alliance.repository'];
+
+/** @var AllianceBuildingRepository */
+$buildingRepository = $app['etoa.alliance.building.repository'];
+
+/** @var AllianceTechnologyRepository */
+$technologyRepository = $app['etoa.alliance.technology.repository'];
+
+if (isset($_GET['alliance_id'])) {
+	$id = $_GET['alliance_id'];
+}
+if (isset($_GET['id'])) {
+	$id = $_GET['id'];
+}
+
+if (isset($_POST['info_save']) && $_POST['info_save'] != "") {
+	saveInfo($repository, $id);
+} elseif (isset($_POST['member_save']) && $_POST['member_save'] != "") {
+	saveMembers($repository);
+} elseif (isset($_POST['bnd_save']) && $_POST['bnd_save'] != "") {
+	saveDiplomacy($repository);
+} elseif (isset($_POST['res_save']) && $_POST['res_save'] != "") {
+	saveResources($repository, $id);
+} elseif (isset($_POST['buildings']) && $_POST['buildings'] != "") {
+	buildings($buildingRepository, $id);
+} elseif (isset($_POST['techs']) && $_POST['techs'] != "") {
+	technologies($technologyRepository, $id);
+}
+edit($repository, $buildingRepository, $technologyRepository, $id);
+
+function saveInfo(AllianceRepository $repository, int $id)
+{
+	global $twig;
+
+	//  Bild löschen wenn nötig
+	if (isset($_POST['alliance_img_del'])) {
+		$picture = $repository->getPicture($id);
+		if ($picture != null) {
```

Comment:
if possible can we use strict comparison?

Reply:
1fbdf38 solved